tools/swift-format/Sources/Rules/UseSynthesizedInitializer.swift

```diff
@@ -12,5 +12,110 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#initializers-2
 public final class UseSynthesizedInitializer: SyntaxLintRule {
+  public override func visit(_ node: StructDeclSyntax) {
+    var storedProperties: [VariableDeclSyntax] = []
+    var initializers: [InitializerDeclSyntax] = []
 
+    for member in node.members.members {
+      // Collect all stored variables into a list
+      if let varDecl = member.decl as? VariableDeclSyntax {
+        guard let modifiers = varDecl.modifiers else {
+          storedProperties.append(varDecl)
+          continue
+        }
+        guard !modifiers.has(modifier: "static") else { continue }
+        storedProperties.append(varDecl)
+      // Collect any possible redundant initializers into a list
+      } else if let initDecl = member.decl as? InitializerDeclSyntax {
+        guard initDecl.modifiers == nil ||
+              initDecl.modifiers!.has(modifier: "internal") else { continue }
+        guard initDecl.optionalMark == nil else { continue }
+        guard initDecl.throwsOrRethrowsKeyword == nil else { continue }
+        initializers.append(initDecl)
+      }
+    }
+
+    for initializer in initializers {
+      guard matchesPropertyList(parameters: initializer.parameters.parameterList,
+                                properties: storedProperties) else { continue }
+      guard matchesAssignmentBody(variables: storedProperties,
+                                  initBody: initializer.body) else { continue }
+      diagnose(.removeRedundantInitializer, on: initializer)
+    }
+  }
+
+  // Compares initializer parameters to stored properties of the struct
+  func matchesPropertyList(parameters: FunctionParameterListSyntax,
+                            properties: [VariableDeclSyntax]) -> Bool {
+    guard parameters.count == properties.count else { return false }
+    for (idx, parameter) in parameters.enumerated() {
+
+      guard let paramId = parameter.firstName, parameter.secondName == nil else { return false }
+      guard let paramType = parameter.type else { return false }
+
+      let property = properties[idx]
+      let propertyId = property.identifier
+      guard let propertyType = property.type else { return false }
+
+      // Sythesized initializer only keeps default argument if the declaration uses 'var'
+      if property.letOrVarKeyword.tokenKind == .varKeyword {
+        if let initializer = property.initializer {
+          guard let defaultArg = parameter.defaultArgument else { return false }
+          guard initializer.value.description == defaultArg.value.description else { return false }
+        }
+      }
+
+      if propertyId.identifier.text.trimmingCharacters(in: .whitespaces) !=
```

Comment:
Do you need to trim whitespace off of these? I would expect the identifier text to not include any.

The only place where I would think it would be an issue is when you call `description`, since that might include leading/trailing trivia.